vllm/entrypoints/openai/api_server.py

```diff
@@ -488,6 +498,48 @@ async def create_score_v1(request: ScoreRequest, raw_request: Request):
     return await create_score(request, raw_request)
 
 
+TASK_HANDLERS = {
+    "generate": {
+        "messages": (ChatCompletionRequest, create_chat_completion),
+        "default": (CompletionRequest, create_completion),
+    },
+    # should `embed` be a pooling task by default?
+    "embed": {
```

Comment:
All pooling tasks also support the "pooling" endpoint.

Reply:
@DarkLight1337 Understood. I can't think of a clean way to expose `pooling` based on `task` as I've done her.

We could take an optional arg `return_pooling` which we handle for pooling tasks.

I'm happy to handle an additional request, or leave it out for now. The SageMaker entrypoint doesn't need to be fully featured for most use cases, especially since the vast majority will use `generate`.

Let me know your thoughts.

Reply:
Instead of handling by `task`, perhaps it would be better to select the handler based on the request type?

Reply:
For example, we can call `model_validate` on a union of the possible types, then check which type the request resolves to.

Reply:
I had this originally, but unfortunately some types are not mutually exclusive. e.g. The pooling ones are exactly equal to the Embedding ones:
```py
PoolingCompletionRequest = EmbeddingCompletionRequest
PoolingChatRequest = EmbeddingChatRequest
```
If the `/v1/embeddings` outputs are always equal to `/pooling` then we're fine, but I'm not sure if this is the case?

In the case of `score` we can definitely do this though, since they have different request structures.

Reply:
> If the `/v1/embeddings` outputs are always equal to `/pooling` then we're fine, but I'm not sure if this is the case?

The output schema is a bit different (`EmbeddingResponse` vs `PoolingResponse`).

> I had this originally, but unfortunately some types are not mutually exclusive.

Hmm... let's just use `EmbeddingResponse` for `--task embed` then. The other pooling tasks ("reward" and "classify") can use `PoolingResponse`.

Reply:
Great - let me know if you're happy on this front and we can resolve this point, as well as if there are any other issues or areas for improvement!

FYI I tried using Pydantic to parse the Union types with `TypeAdapter.validate_python` but `mypy` was very unhappy so left the current pattern 😅 Potentially an area for improvement in future, although I think the current solution is quite pragmatic and readable.

I will hopefully get some time to test this on SageMaker in the next couple of weeks. If you do approve, I'll still hold off merging until I get a chance to build and test it on AWS to make sure there's nothing missing key on the integration.